src/core/modules/BasedAppsManager.sol

```diff
@@ -43,12 +42,42 @@ contract BasedAppsManager is IBasedAppManager {
 
     /// @notice Function to update the metadata URI of the Based Application
     /// @param metadataURI The new metadata URI
-    function updateBAppMetadataURI(
-        string calldata metadataURI
-    ) external onlyRegisteredBApp {
+    function updateBAppMetadataURI(string calldata metadataURI) external {
+        _onlyRegisteredBApp(CoreStorageLib.load());
         emit BAppMetadataURIUpdated(msg.sender, metadataURI);
     }
 
+    function updateBAppsTokens(
+        ICore.TokenConfig[] memory tokenConfigs
+    ) external {
+        CoreStorageLib.Data storage s = CoreStorageLib.load();
+
+        _onlyRegisteredBApp(s);
+
+        uint32 requestTime = uint32(block.timestamp);
+
+        address token;
+        ICore.SharedRiskLevel storage tokenData;
+        ProtocolStorageLib.Data storage sp = ProtocolStorageLib.load();
+
+        for (uint256 i = 0; i < tokenConfigs.length; ) {
+            token = tokenConfigs[i].token;
+            tokenData = s.bAppTokens[msg.sender][token];
+            // Update current value if the previous effect time has passed
+            if (requestTime > tokenData.effectTime) {
+                tokenData.currentValue = tokenData.pendingValue;
+            }
```

Comment:
the thing is if you update beta suddenly upwards, lots of over obligations may stop making sense... Because of `OBLIGATION_TIMELOCK_PERIOD` it will be hard to fix on time.
However the bApp will probably not want to wreck itself so might be fine.